packages/caliper-core/test/worker/rate-control/recordRate.js

```diff
@@ -63,42 +74,262 @@ describe('RecordRate controller', () => {
+        });
+
+
+        it('should export to binary big endian format when output format is BIN_BE', async () => {
+            const msgContent = { ...defaultMsgContent };
+            msgContent.rateControl.opts.outputFormat = 'BIN_BE';
+            const testMessage = new TestMessage('test', [], msgContent);
+            const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+
+            const mockController = {
+                end: sinon.stub().resolves(),
+                applyRateControl: sinon.stub().resolves(),
+            };
+            controller.recordedRateController.controller = mockController;
+
+            const exportToTextSpy = sinon.spy(controller, '_exportToText');
+            const exportToBinaryLittleEndianSpy = sinon.spy(controller, '_exportToBinaryLittleEndian');
+            const exportToBinaryBigEndianSpy = sinon.spy(controller, '_exportToBinaryBigEndian');
+
+            await controller.end();
+
+            sinon.assert.notCalled(exportToTextSpy);
+            sinon.assert.notCalled(exportToBinaryLittleEndianSpy);
+            sinon.assert.calledOnce(exportToBinaryBigEndianSpy);
+            sinon.assert.notCalled(logger.error);
+
+            exportToTextSpy.restore();
+            exportToBinaryLittleEndianSpy.restore();
+            exportToBinaryBigEndianSpy.restore();
+        });
+
+        it('should throw an error if pathTemplate is undefined', () => {
+            const msgContent = { ...defaultMsgContent };
+            msgContent.rateControl.opts.pathTemplate = undefined;
+            const testMessage = new TestMessage('test', [], msgContent);
+
+            (() => {
+                RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+            }).should.throw('The path to save the recording to is undefined');
+        });
+    });
+
+    describe('Rate Control', () => {
+        it('should apply rate control to the recorded rate controller', async () => {
+            const testMessage = new TestMessage('test', [], defaultMsgContent);
+            const rateController = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+            const mockRate = MockRate.createRateController();
+            mockRate.reset();
+            mockRate.isApplyRateControlCalled().should.equal(false);
+            await rateController.applyRateControl();
+            mockRate.isApplyRateControlCalled().should.equal(true);
+        });
+
+        it('should initialize records array if the number of transactions is provided', () => {
+            const msgContent = {...defaultMsgContent };
+
+            const testMessage = new TestMessage('test', [], msgContent);
+            sinon.stub(testMessage, 'getNumberOfTxs').returns(5);
+
+            const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+
+            controller.records.should.be.an('array').that.has.lengthOf(5);
```

Comment:
Here you have written a test that passes when it fact there is a bug here. The length should be 6, because the total possible values for the number of transactions submitted is 0-5.